app/code/Magento/Customer/Block/CustomerScopeData.php

```diff
@@ -50,4 +54,37 @@ public function getWebsiteId()
     {
         return (int)$this->_storeManager->getStore()->getWebsiteId();
     }
+
+    /**
+     * @return array
+     */
+    public function getInvalidationRules()
+    {
+        return [
+            '*' => [
+                'Magento_Customer/js/invalidation-processor' => [
+                    'invalidationRules' => [
+                        'website-rule' => [
+                            'Magento_Customer/js/invalidation-rules/website-rule' => [
+                                'scopeConfig' => [
+                                    'websiteId' => $this->getWebsiteId(),
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+            ],
+        ];
+    }
+
+    /**
+     * Get the invalidation rules json encoded
+     *
+     * @return bool|string
+     * @throws \InvalidArgumentException
+     */
+    public function getSerializedInvalidationRules()
+    {
+        return $this->serializer->serialize($this->getInvalidationRules());
```

Comment:
Yeah something like `escapeHtml` would be a nice option here also thought about that. It's above this PR but I think some "common" way to do this through templates would be amazing.